dashboard/app/models/user.rb

```diff
@@ -884,6 +887,28 @@ def in_progress_and_completed_scripts
     end
   end
 
+  # This method is meant to return a bunch of data about a user's recent courses.
+  # Right now, it just looks at recent scripts, but we expect it to change in the
+  # very near future
+  # TODO: Once this data is more real, we should be writing tests for it.
+  def recent_courses
+    in_progress_and_completed_scripts.map do |script|
+      script_id = script[:script_id]
+      script_name = Script.get_from_cache(script_id)[:name]
+      {
+        id: script_id,
+        script_name: script_name,
+        courseName: data_t_suffix('script.name', script_name, 'title'),
+        description: data_t_suffix('script.name', script_name, 'description_short'),
+        # TODO: This can probably be generated on the client. If it can, we can also
+        # get rid of include Rails.application.routes.url_helpers
+        link: script_url(script_id),
+        image: "",
+        assignedSections: []
```

Comment:
They are used by React components (which will presumably warn if they're empty). I expect the shape of this data to still change a bit in the nearish future.